### PR TITLE
Add to cart.

### PR DESCRIPTION
**Pull request by franzsalvador**

### CODE REVIEW

**Comment by franzsalvador** [comment]
jam 3 5

File: main.js [hunk 1]
```diff
@@ -239,3 +254,10 @@ function view(viewName) {
     renderDetails.setAttribute('class', 'hidden')
   }
 }
+
+function buildCart(cartList) {
```
Comment (thebearingedge) [inline main.js]: Are you using this function for anything?

File: main.js [hunk 2]
```diff
@@ -151,6 +155,17 @@ function renderApp(app) {
...
     renderDetails.appendChild(detailsView)
+    var btn = document.getElementById('button')
+    var cartBadge = document.querySelector('span')
+    btn.addEventListener('click', function (event) {
+      app.cart.items.push(app.details.item)
+      cartBadge.textContent = '(' + (app.cart.items.length) + ')'
+      app.view = 'catalog'
+    })
+  }
+  else if (app.view === 'cart') {
```
Comment (thebearingedge) [inline main.js]: We'll be dealing with the cart view in a couple issues. You can remove this `if` for now.

File: main.js [hunk 3]
```diff
@@ -87,11 +87,15 @@ var app = {
...
     item: null
+  },
+  cart: {
+    items: []
   }
 }
 
 var renderCatalog = document.querySelector("[data-view='catalog']")
 var renderDetails = document.querySelector("[data-view='details']")
+var renderCart = document.querySelector("[data-view='cart']")
```
Comment (thebearingedge) [inline main.js]: We'll be dealing with the cart view in a couple issues. You can remove this variable for now.

File: index.html [hunk 4]
```diff
@@ -8,10 +8,16 @@
...
+    <div id="cart">
+      <a href="#">Cart <span class="badge">(0)</span></a>
+    </div>
+    <div>
+      <h1></h1>
+    </div>
     <div id="app">
       <div data-view="catalog"></div>
       <div data-view="details"></div>
+      <div data-view="cart"></div>
```
Comment (thebearingedge) [inline index.html]: We'll be dealing with the cart view in a couple issues. You can remove this `data-view` for now.

**Review by thebearingedge — CHANGES_REQUESTED** [review]
A little clean up and you're good to go.

File: main.js (hunk 1 above)
Comment (franzsalvador) [reply main.js]: I think I may have misunderstood this one. 
Define a function that takes a `cart` object and returns a DOM tree containing the count of items in the cart. Is that what this one was asking?

Comment (thebearingedge) [reply main.js]: More like:

```js
function buildCart(cart) {
  var $cart = document.createElement('span')
  $cart.classList.add('badge')
  $cart.textContent = 'Cart (' + cart.items.length + ')'
  return $cart
}
```

Looks like you were just updating the `textContent` of an element already in the page, so this function isn't needed.

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**